### DISCUSSION TITLE
CORS resource not load

### DISCUSSION THREAD

**Discussion by loriszigliani** [mozilla/hubs#3926] [Hubs Cloud Questions] [answered]
Hello everybody.

i installed cloud hubs on aws, everything works perfectly.
Through a .spoke file I have inserted the project in spoke on cloud hubs and I can view it but the videos from vimeo, twitch, and a fatlab model are not displayed giving me an error.
i tried to understand how CORS works but i didn't understand. what should I do?

**Comment by robinkwilson** [comment]
Hello! Yes, CORS is tricky.

Fill in settings in the Admin Panel (and add additional domains if you want):

1. Go to your hub's main domain > Admin Panel > Server Settings > Advanced tab
2. In your "Allowed CORS Origin" field:

    https://{main domain}.com, https://{link domain}.link, {main domain}.com, {link domain}.link, https://{internal domain}.com, {internal domain}.com, https://vimeo.com/, https://twitch.tv/, {fatlab url}, {any other domains you want to approve}

^^ Here's what worked for me. REMEMBER to change {main domain}, {internal domain}, {link domain}, and the domain extension (.com, .link) to your hub's.
(I added the internal domain using the same standard (my stack used Recipe 1 so it didn't have a different internal domain)

If you have any problems follow the answer outlined in this github discussion to get unstuck: https://github.com/mozilla/hubs/discussions/2890

**Comment by edap** [comment, marked as answer]
I can not upload assets to spoke on https://hubs.mozilla.com/spoke/.
The error is:
```

o: Upload failed:
  Cause:
    Network Error: Unknown Status. Unknown Error. Possibly a CORS error.
    r errors.js:29
    o errors.js:37
    upload Api.js:1054
    r helpers.js:59
EditorContainer.js:860:14
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://elastic-orc.reticulum.io/api/v1/media. (Reason: CORS request did not succeed).
```

**Reply by brianpeiris** [reply]
Hi @edap This was probably due to the downtime we experienced today. Spoke should be working now.

**Reply by edap** [reply]
Hello Brian, thanks for your reply. The problem is happening right now, is there a service where is possible to check the downtime status?

**Reply by brianpeiris** [reply]
@edap We don't currently have a status page setup. I'm not able to reproduce any issues with asset uploads in Spoke. I'd suggest you create a separate discussion or bug report with your details, since this discussion was primarily about Hubs Cloud, not hubs.mozilla.com.

**Comment by edap** [comment]
I see. Now it is working again, if it happens again I will open the issue in the page you have suggested, thanks.
